sklearnex/basic_statistics/incremental_basic_statistics.py

```diff
@@ -0,0 +1,177 @@
+
+        second_order_raw_moment : ndarray of shape (n_features,)
+            Second order moment of each feature over all samples.
+    """
+
+    _onedal_incremental_basic_statistics = staticmethod(onedal_IncrementalBasicStatistics)
+
+    def __init__(self, result_options="all", batch_size=None):
+        if result_options == "all":
+            self.result_options = (
+                self._onedal_incremental_basic_statistics.get_all_result_options()
+            )
+        else:
+            self.result_options = result_options
+        self._need_to_finalize = False
+        self.batch_size = batch_size
+
+    def _get_onedal_result_options(self, options):
+        if isinstance(options, list):
+            onedal_options = "|".join(self.result_options)
+        else:
+            onedal_options = options
+        assert isinstance(onedal_options, str)
+        return options
+
+    def _onedal_finalize_compute(self):
+        assert hasattr(self, "_onedal_estimator")
+        self._onedal_estimator.finalize_compute()
+        self._need_to_finalize = False
+
+    def _onedal_partial_compute(self, X, weights):
+        onedal_params = {
+            "result_option": self._get_onedal_result_options(self.result_options),
+            "method": "by_default",
+        }
+        if not hasattr(self, "_onedal_estimator"):
+            self._onedal_estimator = self._onedal_incremental_basic_statistics(
+                **onedal_params
+            )
+        self._onedal_estimator.partial_compute(X, weights)
+        self._need_to_finalize = True
+
+    def __getattr__(self, attr):
+        result_options = self.__dict__["result_options"]
+        is_statistic_attr = (
+            isinstance(result_options, str) and (attr == result_options)
+        ) or (isinstance(result_options, list) and (attr in result_options))
+        if is_statistic_attr:
+            if self._need_to_finalize:
+                self._onedal_finalize_compute()
+            return getattr(self._onedal_estimator, attr)
+        if attr in self.__dict__:
+            return self.__dict__[attr]
+
+        raise AttributeError(
+            f"'{self.__class__.__name__}' object has no attribute '{attr}'"
+        )
+
+    def partial_fit(self, X, weights=None):
+        """Incremental fit with X. All of X is processed as a single batch.
+
+        Parameters
+        ----------
+        X : array-like of shape (n_samples, n_features)
+            Data for compute, where `n_samples` is the number of samples and
+            `n_features` is the number of features.
+
+        weights : array-like of shape (n_samples,)
+            Weights for compute weighted statistics, where `n_samples` is the number of samples.
+
+        Returns
+        -------
+        self : object
+            Returns the instance itself.
+        """
+        self._onedal_partial_compute(X, weights)
+        return self
+
+    def fit(self, X, weights=None):
```

Comment:
Use support_usm_ndarray decorator from here: https://github.com/intel/scikit-learn-intelex/blob/0e137ab626994800bb021812ab1afbe8c340a165/onedal/_device_offload.py#L62C5-L62C24
```suggestion
    @support_usm_ndarray()
    def fit(self, X, weights=None, queue=None):
```